src/governance/LendingTermOnboarding.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {Governor, IGovernor} from "@openzeppelin/contracts/governance/Governor.sol";
+
+import {Core} from "@src/core/Core.sol";
+import {CoreRoles} from "@src/core/CoreRoles.sol";
+import {GuildToken} from "@src/tokens/GuildToken.sol";
+import {LendingTerm} from "@src/loan/LendingTerm.sol";
+import {VoltGovernor} from "@src/governance/VoltGovernor.sol";
+
+/// @notice Utils to onboard a LendingTerm. Also acts as a LendingTerm factory.
+/// This contract acts as Governor, but users cannot queue arbitrary proposals,
+/// they can only queue LendingTerm onboarding proposals.
+/// When a vote is successful, the LendingTerm onboarding is queued in the Timelock,
+/// where CREDIT holders can veto the onboarding.
+/// Only terms that have been deployed through this factory can be onboarded.
+/// A term can be onboarded for the first time, or re-onboarded after it has been offboarded.
+contract LendingTermOnboarding is VoltGovernor {
+
+    uint256 MIN_DELAY_BETWEEN_PROPOSALS = 7 days;
+    address public immutable lendingTermImplementation;
+    address public immutable guildToken;
+    uint256 public immutable gaugeType;
+    mapping(address=>uint256) public created;
+    mapping(address=>uint256) public lastProposal;
+    
+    constructor(
+        address _lendingTermImplementation,
+        address _guildToken,
+        uint256 _gaugeType,
+        address _core,
+        address _timelock,
+        address _token,
+        uint256 initialVotingDelay,
+        uint256 initialVotingPeriod,
+        uint256 initialProposalThreshold,
+        uint256 initialQuorum
+    )
+        VoltGovernor(
+            _core,
+            _timelock,
+            _token,
+            initialVotingDelay,
+            initialVotingPeriod,
+            initialProposalThreshold,
+            initialQuorum
+        )
+    {
+        lendingTermImplementation = _lendingTermImplementation;
+        guildToken = _guildToken;
+        gaugeType = _gaugeType;
+    }
+
+    /// @notice Create a new LendingTerm and initialize it.
+    function createTerm(LendingTerm.LendingTermParams calldata params) external returns (address) {
```

Comment:
there should be sanity checks here to prevent malicious or incorrect proposals

as an example, interest rate should never be 1000%,

token should have a non zero extcodesize,

if maxDelayBetweenPartialRepay is non zero, its value is bounded within some reasonable amount

Reply:
hm interesting, why not. We can discuss at next meeting

Reply:
done